src/eval.c

```diff
@@ -264,15 +264,27 @@ void scriptingInit(int setup) {
     lctx.lua = lua;
 }
 
+/* Free lua_scripts dict and close lua interpreter. */
+void freeLuaScriptsSync(dict *lua_scripts, lua_State *lua) {
+    dictRelease(lua_scripts);
+    lua_close(lua);
+
+#if defined(__linux__) && !defined(__APPLE__) && !defined(USE_LIBC)
```

Comment:
we need a comment that states that we do that only when Redis itself DOESN'T use libc too.
i.e. when Lua and Redis use different allocators, one won't use the fragmentation holes of the other, and released memory can take long time until it is returned to the OS.